lib/project_types/script/forms/create.rb

```diff
@@ -20,7 +20,11 @@ def ask_extension_point
       end
 
       def ask_name
-        CLI::UI::Prompt.ask(@ctx.message('script.forms.create.script_name'))
+        name = CLI::UI::Prompt.ask(@ctx.message('script.forms.create.script_name'))
+        unless !name.match?(/[^0-9A-Za-z _-]/)
```

Comment:
Any thoughts on reducing this to `unless name.match?(/[0-9A-Za-z _-]/)` ?

Reply:
Sounds good to me, is this updated change what you had in mind?

```        
name = CLI::UI::Prompt.ask(@ctx.message('script.forms.create.script_name'))
unless name.match?(/[^0-9A-Za-z _-]/)
  return name
end
@ctx.abort(@ctx.message('script.forms.create.error.invalid_name'))```

Reply:
I was meaning we could avoid the 'double negation' by also removing the `^` from the character class (the part in square brackets).

But I now realise example was slightly off, it should be something like:

`unless name.match?(/^[0-9A-Za-z _-]*$/)`

(the `^` has a different meaning outside of square brackets – it acts as an anchor).

Reply:
(but looking at the way you've change it that's also fine).

Reply:
I do like `name.match?(/^[0-9A-Za-z _-]*$/)` because of the double negation issue.

```ruby
return name if name.match?(/^[0-9A-Za-z _-]*$/)
```
Reads rather nicely